pkg/controller/subnetcache.go

```diff
@@ -0,0 +1,163 @@
+// Copyright 2017 Cisco Systems, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package controller
+
+import (
+	"fmt"
+	"strings"
+	"sort"
+    "github.com/noironetworks/aci-containers/pkg/apicapi"
+)
+
+func (cont *AciController) SubnetChanged(obj apicapi.ApicObject, aciVrfDn string) {
+	subnetDn := obj.GetAttrStr("dn")
+	subnetIp := obj.GetAttrStr("ip")
+	
+	cont.log.Debug("SubnetChanged for dn: ", subnetDn)
+	if _, ok := cont.apicConn.CachedSubnetDns[subnetDn]; !ok {
+		cont.UpdateSubnetDnCache(subnetDn, subnetIp, aciVrfDn)
```

Comment:
is there any possibility that change notification will called even though there is no change in subnets? in that case you can avoid calling the sync

Reply:
Fixed.